recipes/onednn/meta.yaml

```diff
@@ -0,0 +1,95 @@
+{% set name = "oneDNN" %}
+{% set version = "2.1.2" %}
+# TODO: Put this in conda_build_config.yaml once we support multiple variants
+{% set dnnl_cpu_runtime = "tbb" %}
+
+package:
+  name: {{ name|lower }}-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/oneapi-src/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: cca53231ec99878dc7ef3cf4984525df4691b8174e703b40dd530c50531ecea0
+
+build:
+  number: 0
+
+outputs:
+  - name: {{ name|lower }}-cpu-mutex
```

Comment:
Why do you need this when `onednn-cpu-tbb` is there?

Reply:
`onednn-cpu-tbb` is there to be in line with the packages provided in the `intel` channel. I have added this to be in line how we typically handle implementation selection in conda-forge.

Reply:
I think it's better to be explicit and we really don't have any conventions in conda-forge.

Reply:
Also, it looks like everything can be switched at runtime, so I don't think we need any variants.

Reply:
> everything can be switched at runtime

I'm not sure I understand how that is possible. You can build oneDNN with either OpenMP or TBB but not with both at the same time.

Reply:
Ah, okay. So there are CPU variants. GPU routines and CPU ones can co-exist in the same library here right?

Reply:
Yes, you can specify exactly one CPU and one GPU implementation per build. The current build has no GPU support though.

Reply:
It is only one GPU implementation or any GPU implementation with OpenCL?

Reply:
The README looks like there is the differentiation between "any GPU implementation with OpenCL" and "OpenCL with the DPC++ Compiler" but I have refer here to @napetrov . I'm no GPU expert anymore.

Reply:
Per my understanding oneDNN have support for both OpenCL interfaces and DPC++ interfaces.  But they are part of single build configuration

Reply:
Is this then using `DNNL_GPU_RUNTIME=DPCPP` for the build configuration? This would then make the variant selection for GPUs simpler.